src/main/java/com/minecolonies/coremod/colony/buildings/modules/EntityListModule.java

```diff
@@ -0,0 +1,121 @@
+package com.minecolonies.coremod.colony.buildings.modules;
+
+import com.google.common.collect.ImmutableList;
+import com.minecolonies.api.colony.buildings.modules.AbstractBuildingModule;
+import com.minecolonies.api.colony.buildings.modules.IEntityListModule;
+import com.minecolonies.api.colony.buildings.modules.IPersistentModule;
+
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Class for all buildings that need a list of mobs to toggle for various reasons.
+ */
+public class EntityListModule extends AbstractBuildingModule implements IEntityListModule, IPersistentModule
+{
+    /**
+     * Tag to store the mob list.
+     */
+    private static final String TAG_MOBLIST = "moblist";
+    private static final String TAG_MOB = "mob";
+
+    /**
+     * List of allowed items.
+     */
+    private final Set<ResourceLocation> mobsAllowed = new HashSet<>();
+
+    /**
+     * Unique id of this module.
+     */
+    private final String id;
+
+    /**
+     * Construct a new grouped itemlist module with the unique list identifier.
+     * @param id the list id.
+     */
+    public EntityListModule(final String id)
+    {
+        super();
+        this.id = id;
+    }
+
+    @Override
+    public void deserializeNBT(final CompoundNBT compound)
+    {
+        final ListNBT filterableList = compound.getCompound(id).getList(TAG_MOBLIST, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < filterableList.size(); ++i)
+        {
+            mobsAllowed.add(new ResourceLocation(filterableList.getCompound(i).getString(TAG_MOB)));
```

Comment:
This is not compatible with the previous serialization (that used `mobview`, not `mob`), so it's not backwards compatible and all existing guard towers will lose their current settings.

Either that's a bug by itself, or if you're intentionally breaking compatibility then you might as well go all the way and make it a plain list rather than a list of compounds for efficiency.  (Though that also might require some defensive code for the upgrade case, which this doesn't.)

Reply:
I have to double check this, but I'm breaking it intentionally to lose the values yes. I can use a string list true. I have to load this from a different compound I believe initially to make sure it doesn't throw an exception though. Forgot about that.